### PR TITLE
fix: Update oss charts. Add acceptance tests for helm

### PR DESCRIPTION
**Pull request by xpuska513**
## What
* Include changes from #15957 
* Add acceptance tests for Helm charts
* Fix minor issues
* Include changes from #16160 
* Remove bitnami minio and postgres chart and replace them with own deployments
* Update docs in all subcharts values.yaml files to refer default env variables setup

## Recommended reading order
1. `x.java`
2. `y.python`

## 🚨 User Impact 🚨
Are there any breaking changes? What is the end result perceived by the user? If yes, please merge this PR with the 🚨🚨 emoji so changelog authors can further highlight this if needed.

### CODE REVIEW

File: .github/workflows/gradle.yml
Comment (davinchia) [inline .github/workflows/gradle.yml:692]: nice! I'm guessing we will remove the current kustomize test in a follow up PR?

File: charts/airbyte-pod-sweeper/Chart.lock
Comment (davinchia) [inline charts/airbyte-pod-sweeper/Chart.lock:1]: why remove this?

File: charts/airbyte-pod-sweeper/templates/configmap.yaml
Comment (davinchia) [inline charts/airbyte-pod-sweeper/templates/configmap.yaml:10]: nit : newline

File: charts/airbyte-pod-sweeper/Chart.lock
Comment (davinchia) [reply charts/airbyte-pod-sweeper/Chart.lock:1]: To get rid of the bitnami security warnings?

File: charts/airbyte-server/templates/secrets.yaml
Comment (davinchia) [inline charts/airbyte-server/templates/secrets.yaml:10]: why do we need to add this?

File: charts/airbyte-server/templates/service.yaml
Comment (davinchia) [inline charts/airbyte-server/templates/service.yaml:16]: nit: new line

File: charts/airbyte-temporal/templates/_helpers.tpl
Comment (davinchia) [inline charts/airbyte-temporal/templates/_helpers.tpl:51]: Curious, what does this render to?

Is it:
airbyte: temporal?

File: charts/airbyte-webapp/templates/secrets.yaml
Comment (davinchia) [inline charts/airbyte-webapp/templates/secrets.yaml:10]: same question here - why do we need this?

File: charts/airbyte/Chart.yaml.test
Comment (davinchia) [inline charts/airbyte/Chart.yaml.test:64]: nit: new line

File: charts/airbyte/Chart.yaml.test
Comment (davinchia) [inline charts/airbyte/Chart.yaml.test:1]: why do we need a `Chart.yaml.test`?

File: charts/airbyte/values.yaml
Comment (davinchia) [inline charts/airbyte/values.yaml:10]: what's happening to these values?

File: charts/airbyte/Chart.yaml.test
Comment (davinchia) [reply charts/airbyte/Chart.yaml.test:1]: Ah I see - we are using this in the test script

File: tools/bin/acceptance_test_kube_helm.sh
Comment (davinchia) [inline tools/bin/acceptance_test_kube_helm.sh:24]: Why do we need to use a test one?

File: tools/bin/acceptance_test_kube_helm.sh
Comment (davinchia) [inline tools/bin/acceptance_test_kube_helm.sh:89]: nit: newline

**Review by davinchia — COMMENTED** [review]
Nice work! I'm curious about some of the pieces here and have questions to make sure I understand the changes before we merge this in.

Tips for the future - will be nice to fill out the PR description to make clear what the changes are & how to review. Will also be nice to split into smaller PRs if there is alot happening that isn't related :)

File: tools/bin/acceptance_test_kube_helm.sh
Comment (xpuska513) [reply tools/bin/acceptance_test_kube_helm.sh:24]: To support local testing + acceptance

File: .github/workflows/gradle.yml
Comment (xpuska513) [reply .github/workflows/gradle.yml:692]: I think we can, but let's discuss it on ours todays sync-up

File: charts/airbyte-pod-sweeper/Chart.lock
Comment (xpuska513) [reply charts/airbyte-pod-sweeper/Chart.lock:1]: Leftovers from testing the charts

File: charts/airbyte-server/templates/secrets.yaml
Comment (xpuska513) [reply charts/airbyte-server/templates/secrets.yaml:10]: If you'll add secrets to your subchart, we need to make sure that secrets exists before deploying the app itself. Otherwise it'll error about secrets not found in the namespace

File: charts/airbyte-temporal/templates/_helpers.tpl
Comment (xpuska513) [reply charts/airbyte-temporal/templates/_helpers.tpl:51]: No, it'll be 
```
app.kubernetes.io/name: temporal
```
It's a proposed changes from a contributor in #15957 to make helm remove all the stuff it deployed(it works)

File: charts/airbyte-webapp/templates/secrets.yaml
Comment (xpuska513) [reply charts/airbyte-webapp/templates/secrets.yaml:10]: Same answer as above, to be sure that we have secrets for the deployment before creating the deployment itself(so it'll find those if specified in values.yaml)

File: charts/airbyte/Chart.yaml.test
Comment (xpuska513) [reply charts/airbyte/Chart.yaml.test:1]: Local testing + acceptance testing

File: charts/airbyte/values.yaml
Comment (xpuska513) [reply charts/airbyte/values.yaml:10]: I'll work on the documentation in my next PR, I just left it blank, so the default value for secrets will be used

File: tools/bin/acceptance_test_kube_helm.sh
Comment (xpuska513) [reply tools/bin/acceptance_test_kube_helm.sh:24]: I did that on purpose, before testing make the .test chart.yaml main one and after reset everything to normal.
So we can run this script both locally and on CI server

**Comment by xpuska513** [comment]
BTW, @davinchia , right now looking why the acceptance tests failed, since I did not changed that much in terms of actual functionality of helm charts(only added a few labels).
Probably everything failing due to changed labeling approach, I'll dig into source code and let you know about any results.

**Comment by supertopher** [comment]
@xpuska513 can you ping me again when you want another review?  

I'm mostly looking for obvious errors with my lack of in depth understanding here.  
happy to sit out if you want Davin/Peter to give more detailed feedback

**Comment by gmontanola** [comment]
https://github.com/airbytehq/airbyte/blob/67ceb9ff7f88ff9c04958232a2c6baa592cad7ab/charts/airbyte-webapp/templates/service.yaml#L5

I think we should add `{{ $fullname }}` here, because the Ingress [expects this name](https://github.com/airbytehq/airbyte/blob/67ceb9ff7f88ff9c04958232a2c6baa592cad7ab/charts/airbyte-webapp/templates/ingress.yaml#L53). The same happens on `server`.

Thanks a lot for all the work you're doing with Helm Charts @xpuska513 💯

File: charts/airbyte-pod-sweeper/templates/configmap.yaml
Comment (davinchia) [reply charts/airbyte-pod-sweeper/templates/configmap.yaml:11]: isn't this still here?

File: charts/airbyte-server/templates/secrets.yaml
Comment (davinchia) [reply charts/airbyte-server/templates/secrets.yaml:10]: got it. how come this wasn't erroring out before?

File: charts/airbyte-temporal/templates/_helpers.tpl
Comment (davinchia) [reply charts/airbyte-temporal/templates/_helpers.tpl:51]: understand.

Comment (davinchia) [reply charts/airbyte-temporal/templates/_helpers.tpl:51]: as long as this is ok in Cloud.

File: charts/airbyte-pod-sweeper/Chart.lock
Comment (davinchia) [reply charts/airbyte-pod-sweeper/Chart.lock:1]: What do you mean?

File: tools/bin/acceptance_test_kube_helm.sh
Comment (davinchia) [reply tools/bin/acceptance_test_kube_helm.sh:24]: got it. how is someone supposed to apply this then?

Comment (davinchia) [reply tools/bin/acceptance_test_kube_helm.sh:24]: and how does this setting this to .test affect publishing?

**Comment by davinchia** [comment]
@xpuska513 we sometimes see this error in the test. I'm not sure why, @tkorenko was looking into this as part of https://github.com/airbytehq/airbyte-cloud/pull/2032/files. We might need to do the same set up here (and in the OSS acceptance tests in general)

**Comment by davinchia** [comment]
@xpuska513 this is actually great stuff! We need to make sure the env vars match the current Kustomize env vars to make sure the Helm charts are a drop-in-replacements for Kustomise. E.g. this is the [dev](https://github.com/airbytehq/airbyte/blob/master/kube/overlays/dev/.env) kustomise env vars. Let's pair on this tmrw.

**Comment by xpuska513** [comment]
@gmontanola, I changed the service name, to the one used in the webapp service defenition + changed the service name so it's not hardcoded now

**Comment by davinchia** [comment]
@xpuska513 what else is remaining in this PR? Does it need another review? Want to be aware of leaving this hanging for too long.

**Comment by xpuska513** [comment]
I think we can merge it in, since it's too big anyways. If there's something left it's probably minor thing and won't break the charts itself

**Review by davinchia — APPROVED** [review]
This is a big PR so let's merge this in for now and iterate on some pieces.

**Pull request merged**
